src/ray/core_worker/experimental_mutable_object_provider.h

```diff
@@ -165,18 +162,19 @@ class MutableObjectProvider {
 
   // Creates a function for each object. This function waits for changes on the object and
   // then sends those changes to a remote node via RPC.
-  std::function<std::shared_ptr<MutableObjectReaderInterface>(const NodeID &node_id)>
+  std::function<std::shared_ptr<MutableObjectReaderInterface>(
+      const NodeID &node_id, rpc::ClientCallManager &client_call_manager)>
       raylet_client_factory_;
-  // Context in which the application looks for local changes to mutable objects and sends
-  // the changes to remote nodes via the network.
-  instrumented_io_context io_service_;
-  // Manages RPCs for inter-node communication of mutable objects.
-  boost::asio::io_service::work io_work_;
-  // Manages outgoing RPCs that send mutable object changes to remote nodes.
-  std::unique_ptr<rpc::ClientCallManager> client_call_manager_;
-  // Thread that waits for local mutable object changes and then sends the changes to
-  // remote nodes via the network.
-  std::thread io_thread_;
+  // Manage RPCs for inter-node communication of mutable objects.
```

Comment:
Can you add a comment explaining why we need to store a vector of threads instead of just one?

Also, I don't this comment is accurate; doesn't `work` should keep the event loop alive, not "manage RPCs"?

Reply:
Good catch! Done.